src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -725,6 +725,9 @@ private TypeSymbol GetTypeOfTypeDef(TypeDefinitionHandle typeDef, out bool isNoP
 
                     switch (allowedRequiredModifierType)
                     {
+                        case AllowedRequiredModifierType.System_Runtime_InteropServices_InOrOutAttribute:
```

Comment:
@333fred Aleksey spotted that we're modifying the same code in different feature branches. Would you be able to make the same change to MetadataDecoder.cs as [here](https://github.com/dotnet/roslyn/pull/43275/files#diff-2231239cabf144e639ea18f4f0ef8627) so that we don't need to worry about merging later on?

For init-only/records, `DecodeModifiersOrThrow` needs to output `out AllowedRequiredModifierType requiredModifiersFound` instead of a `bool`, and make `AllowedRequiredModifierType` be flags so they can be combined. The few callers that consume the output are easily updated.

Let me know it's that would be okay. We could also just wait and merge when the time comes (your changes to this file don't look too bad, so I'm not very worried). Thanks

Reply:
I can take a look at this later. I'm trying to get this in today or tomorrow and I'd prefer to keep these changes isolated.

---
In reply to: [411879270](https://github.com/dotnet/roslyn/pull/43355#discussion_r411879270) [](ancestors = 411879270)

Reply:
I've unified this.

---
In reply to: [412402730](https://github.com/dotnet/roslyn/pull/43355#discussion_r412402730) [](ancestors = 412402730,411879270)